pkg/kn/commands/revision_list.go

```diff
@@ -36,6 +37,9 @@ func NewRevisionListCommand(p *KnParams) *cobra.Command {
 				return err
 			}
 			namespace := cmd.Flag("namespace").Value.String()
+			if getAllNamespaces {
+				namespace = ""
+			}
```

Comment:
How about
```
if getAllNamespaces {
  namespace := ""
} else {
  namespace := cmd.Flag("namespace").Value.String()
}
```
as `all-namespaces` take precedence over `namespace` so we check for it first.

same goes for service list

Reply:
@navidshaikh `:=` would define it inside the if block, so it wont be accessible below.

Reply:
yess.. 👍